frontend/packages/kubevirt-plugin/src/components/modals/vm-description-modal/vm-description-modal.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react';
+import { TextArea } from '@patternfly/react-core';
+
+import { HandlePromiseProps, withHandlePromise } from '@console/internal/components/utils';
+import {
+  createModalLauncher,
+  ModalTitle,
+  ModalBody,
+  ModalSubmitFooter,
+  ModalComponentProps,
+} from '@console/internal/components/factory';
+import { k8sPatch } from '@console/internal/module/k8s';
+
+import { VMLikeEntityKind } from '../../../types';
+import { getDescription, getVmLikeModel } from '../../../selectors/selectors';
+import { getUpdateDescriptionPatches } from '../../../k8s/patches/vm/vm-patches';
+
+import './_vm-description-modal.scss';
+
+export const VmDescriptionModal = withHandlePromise((props: VMDescriptionModalProps) => {
+  const { vmLikeEntity, inProgress, errorMessage, handlePromise, close, cancel } = props;
+
+  const [description, setDescription] = React.useState(getDescription(vmLikeEntity));
+
+  const submit = (e) => {
+    e.preventDefault();
+
+    const patches = getUpdateDescriptionPatches(vmLikeEntity, description);
+    if (patches.length === 0) {
+      close();
+    } else {
+      const promise = k8sPatch(getVmLikeModel(vmLikeEntity), vmLikeEntity, patches);
+      handlePromise(promise).then(close); // eslint-disable-line promise/catch-or-return
+    }
+  };
+
+  return (
+    <form onSubmit={submit} className="modal-content">
+      <ModalTitle>Edit Description</ModalTitle>
+      <ModalBody>
+        <TextArea
+          className="kubevirt-vm-description-modal__description"
+          value={description}
+          onChange={(d) => setDescription(d)}
+          aria-label="description text area"
+        />
+      </ModalBody>
+      <ModalSubmitFooter
+        errorMessage={errorMessage}
+        inProgress={inProgress}
+        submitText="Save"
+        cancel={cancel}
+      />
+    </form>
+  );
+});
+
+export type VMDescriptionModalProps = HandlePromiseProps &
+  ModalComponentProps & {
+    vmLikeEntity: VMLikeEntityKind;
+  };
+
+export const vmDescriptioneModal = createModalLauncher(VmDescriptionModal);
```

Comment:
Are we going to have a dedicated modal component for updating every single logical property?

Maybe we could create some generic modal used for updating simple, text-based properties instead?

Reply:
I would try to make it generic once we have more similar implementations